src/QsCompiler/Compiler/Templates/QirDriverCpp.tt

```diff
@@ -1,16 +1,278 @@
+                    WriteLine($"    vector<char> {arg.Name}AsCharVector;");
+                    WriteLine($"    TranslateVector<PauliId, char>({arg.Name}, {arg.Name}AsCharVector, TranslatePauliToChar);");
+                    WriteLine($"    unique_ptr<InteropArray> {arg.Name}Array = CreateInteropArray({arg.Name}AsCharVector);");
+                    break;
+                case DataType.RangeType:
+                    WriteLine($"    vector<InteropRange*> {arg.Name}Vector;");
+                    WriteLine($"    TranslateVector<RangeTuple, InteropRange*>({arg.Name}, {arg.Name}Vector, TranslateRangeTupleToInteropRangePointer);");
+                    WriteLine($"    unique_ptr<InteropArray> {arg.Name}Array = CreateInteropArray({arg.Name}Vector);");
+                    break;
+                case DataType.ResultType:
+                    WriteLine("    // Create an interop array of Result values.");
+                    WriteLine($"    unique_ptr<InteropArray> {arg.Name}Array = CreateInteropArray({arg.Name});");
+                    break;
+                case DataType.StringType:
+                    WriteLine("    // Create an interop array of String values.");
+                    WriteLine($"    vector<const char *> {arg.Name}BufferVector;");
+                    WriteLine($"    TranslateVector<string, const char*>({arg.Name}, {arg.Name}BufferVector, TranslateStringToCharBuffer);");
+                    WriteLine($"    unique_ptr<InteropArray> {arg.Name}Array = CreateInteropArray({arg.Name}BufferVector);");
+                    break;
+            }
+            break;
+    }
+} #>
+    // Redirect the simulator output from std::cout if the --simulation-output option is present.
+    ostream* simulatorOutputStream = &cout;
+    ofstream simulationOutputFileStream;
+    if (!simulationOutputFileOpt->empty())
+    {
+        simulationOutputFileStream.open(simulationOutputFile);
+        SetOutputStream(simulationOutputFileStream);
+        simulatorOutputStream = &simulationOutputFileStream;
+    }
+
+    // Run simulation and write the output of the operation to the corresponding stream.
+    <#= entryPointOperation.Name #>(
+<# for (int i = 0; i < entryPointOperation.InteropArguments.Count; i++) {
+    var arg = entryPointOperation.InteropArguments[i];
+    switch (arg.Type) {
+        case DataType.PauliType:
+            Write($"        {arg.Name}AsCharValue");
+            break;
+        case DataType.RangeType:
+            Write($"        {arg.Name}Value");
+            break;
+        case DataType.StringType:
+            Write($"        {arg.Name}.c_str()");
+            break;
+        case DataType.ArrayType:
+            Write($"        {arg.Name}Array.get()");
+            break;
+        default:
+            Write($"        {arg.Name}");
+            break;
+    }
+    if (i < entryPointOperation.InteropArguments.Count-1 ) {
+        WriteLine(",");
+    }
+} #>
+
+);
```

Comment:
Nit: there is a blank line before this closing parenthesis and it is not indented.

Reply:
this resolves to the correct spacing in cpp file